code/modules/mob/living/carbon/need_procs.dm

```diff
@@ -0,0 +1,129 @@
+/mob/living/carbon/proc/print_happiness()
+	var/msg = "<span class='info'>*---------*\n<EM>Your current mood</EM>\n"
+	for(var/i in events)
+		var/datum/happiness_event/event = events[i]
+		msg += event.description
+	to_chat(src, msg)
+
+/mob/living/carbon/proc/update_happiness()
+	var/old_happiness = happiness
+	happiness = 0
+	for(var/i in events)
+		var/datum/happiness_event/event = events[i]
+		happiness += event.happiness
+
+	if(happiness > old_happiness)
+		to_chat(src, "<span class='green'>You mood improves.</span>")
+	else if(happiness < old_happiness)
+		to_chat(src, "<span class='warning'>You mood gets worse.</span>")
+	switch(happiness)
+		if(-5000000 to MOOD_LEVEL_SAD4)
+			if(client && hud_used)
+				hud_used.happiness.icon_state = "mood1"
+		if(MOOD_LEVEL_SAD4 to MOOD_LEVEL_SAD3)
+			if(client && hud_used)
+				hud_used.happiness.icon_state = "mood2"
+		if(MOOD_LEVEL_SAD3 to MOOD_LEVEL_SAD2)
+			if(client && hud_used)
+				hud_used.happiness.icon_state = "mood3"
+		if(MOOD_LEVEL_SAD2 to MOOD_LEVEL_SAD1)
+			if(client && hud_used)
+				hud_used.happiness.icon_state = "mood4"
+		if(MOOD_LEVEL_SAD1 to MOOD_LEVEL_HAPPY1)
+			if(client && hud_used)
+				hud_used.happiness.icon_state = "mood5"
+		if(MOOD_LEVEL_HAPPY1 to MOOD_LEVEL_HAPPY2)
+			if(client && hud_used)
+				hud_used.happiness.icon_state = "mood6"
+		if(MOOD_LEVEL_HAPPY2 to MOOD_LEVEL_HAPPY3)
+			if(client && hud_used)
+				hud_used.happiness.icon_state = "mood7"
+		if(MOOD_LEVEL_HAPPY3 to MOOD_LEVEL_HAPPY4)
+			if(client && hud_used)
+				hud_used.happiness.icon_state = "mood8"
+		if(MOOD_LEVEL_HAPPY4 to INFINITY)
+			if(client && hud_used)
+				hud_used.happiness.icon_state = "mood9"
+
+/mob/living/carbon/proc/handle_happiness()
+	switch(happiness)
+		if(-5000000 to MOOD_LEVEL_SAD4)
+			overlay_fullscreen("insanity", /obj/screen/fullscreen/insanity, 3)
+		if(MOOD_LEVEL_SAD4 to MOOD_LEVEL_SAD3)
+			overlay_fullscreen("insanity", /obj/screen/fullscreen/insanity, 2)
+		if(MOOD_LEVEL_SAD3 to MOOD_LEVEL_SAD2)
+			overlay_fullscreen("insanity", /obj/screen/fullscreen/insanity, 1)
+		if(MOOD_LEVEL_SAD2 to MOOD_LEVEL_SAD1)
+			clear_fullscreen("insanity")
+		if(MOOD_LEVEL_SAD1 to MOOD_LEVEL_HAPPY1)
+			clear_fullscreen("insanity")
+		if(MOOD_LEVEL_HAPPY1 to MOOD_LEVEL_HAPPY2)
+			clear_fullscreen("insanity")
+		if(MOOD_LEVEL_HAPPY2 to MOOD_LEVEL_HAPPY3)
+			clear_fullscreen("insanity")
+		if(MOOD_LEVEL_HAPPY3 to MOOD_LEVEL_HAPPY4)
+			clear_fullscreen("insanity")
+		if(MOOD_LEVEL_HAPPY4 to INFINITY)
+			clear_fullscreen("insanity")
```

Comment:
you can use `else` in switches

Reply:
This is placeholder, there will be actual effects for depression and happiness.

Reply:
This is placeholder, there will be actual effects for depression and happiness.